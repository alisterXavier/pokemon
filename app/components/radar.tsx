import ReactApexChart from 'react-apexcharts';

interface Stat {
  stat: {
    name: string;
  };
  base_stat: number;
}

export const RadarChart: React.FC<{ data: Stat[] }> = ({ data }) => {
  const dataStat = data.map(
    (i: { stat: { name: string }; base_stat: number }) => {
      return { name: i.stat.name, stat: i.base_stat };
    }
  );

  var series = [
    {
      name: 'Stats',
      data: dataStat.map((i: { stat: number }) => i.stat),
    },
  ];

  var options = {
    dataLabels: {
      enabled: true,
      background: {
        enabled: true,
        borderRadius: 2,
      },
    },
    chart: {
      toolbar: {
        show: false,
      },
    },
    xaxis: {
      categories: dataStat.map((i: { name: string }) => i.name.toUpperCase()),
    },
    tooltip: {
      theme: 'dark',
    },
  };

  return (
    <ReactApexChart
      options={options}
      series={series}
      type="radar"
      height={350}
      width={450}
    />
  );
};
