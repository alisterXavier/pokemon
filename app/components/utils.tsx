'use client';
import Link from 'next/link';
import styled from 'styled-components';
import loader from '@/public/simple-pokeball.gif';
import Image from 'next/image';

const LinkW = styled(Link)`
  padding: 10px;
  text-transform: uppercase;
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 200px;
  max-width: 300px;
  height: 80px;
  margin: 10px;
  background: #1f1f1f;
  border-radius: 5px;
`;
const Title = styled.h1`
  font-size: 30px;
  text-transform: uppercase;
`;

export const Tile = ({ url, name }: { url: string; name: string }) => {
  return (
    <LinkW href={url} className="type-link">
      {name}
    </LinkW>
  );
};

export const PageTitle = ({ title }: { title: string }) => {
  return (
    <div>
      <Title>{title}</Title>
    </div>
  );
};

export const Loading = () => {
  return (
    <div className="loader">
      <Image src={loader} alt={'loader'} width={200} height={200} />
    </div>
  );
};



export const PokemonImage = ({ url }: { url: string }) => {
  return url ? <Image src={url} alt="" width={100} height={100} /> : <></>;
};
