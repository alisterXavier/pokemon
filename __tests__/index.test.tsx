import '@testing-library/jest-dom';
import { Loading, PageTitle, Tile } from '@/app/components/utils';
import { render, screen } from '@testing-library/react';
import { RadarChart } from '@/app/components/radar';


jest.mock('react-apexcharts', () =>
  jest.fn(() => {
    return null;
  })
);
jest.mock('apexcharts', () => ({
  exec: jest.fn(() => {
    return new Promise((resolve) => {
      resolve('uri');
    });
  }),
}));
const mockData = [
  { stat: { name: 'HP' }, base_stat: 250 },
  { stat: { name: 'Attack' }, base_stat: 110 },
];
describe('Components', () => {
  it('renders a link with the provided name and url', () => {
    const testUrl = '/pokemons?type=fire';
    const testName = 'fire';

    render(<Tile url={testUrl} name={testName} />);
    const linkElement = screen.getByRole('link');

    expect(linkElement).toHaveAttribute('href', testUrl);
    expect(linkElement.textContent).toBe(testName);
  });

  it('renders the provided title', () => {
    const testTitle = 'Search';

    render(<PageTitle title={testTitle} />);
    const titleElement = screen.getByText(testTitle);
    expect(titleElement).toBeInTheDocument();
  });

  it('renders the loader with the correct image', () => {
    render(<Loading />);

    const loaderDiv = screen.getByRole('img', { name: 'loader' });
    expect(loaderDiv).toBeInTheDocument();
    expect(loaderDiv).toHaveAttribute('alt', 'loader');
    expect(loaderDiv).toHaveAttribute('width', '200');
    expect(loaderDiv).toHaveAttribute('height', '200');
  });

  it('processes data and creates chart configuration', () => {
    const { container } = render(<RadarChart data={mockData} />);
    expect(container).toMatchSnapshot();
  });
});
