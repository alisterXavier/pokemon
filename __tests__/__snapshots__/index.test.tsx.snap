// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components processes data and creates chart configuration 1`] = `<div />`;
